net/developer-guide/working-with-outlook-storage-files/working-with-messages-in-a-pst-file/_index.md

```diff
@@ -70,14 +367,93 @@ This articles shows how to Use the [FolderInfo](https://apireference.aspose.com/
 1. Delete messages from the Sent folder by calling the [FolderInfo.DeleteChildItem()](https://apireference.aspose.com/email/net/aspose.email.storage.pst/folderinfo/methods/deletechilditem) method and passing the [MessageInfo.EntryId](https://apireference.aspose.com/email/net/aspose.email.storage.pst/folderinfo/properties/entryid) as a parameter. The following code snippet shows you how to delete messages from a PST file's Sent subfolder.
 
 
+```csharp
+// For complete examples and data files, please go to https://github.com/aspose-email/Aspose.Email-for-.NET
+// The path to the File directory.
+string dataDir = RunExamples.GetDataDir_Outlook() + "Sub.pst";
+
+// Load the Outlook PST file
+PersonalStorage personalStorage = PersonalStorage.FromFile(dataDir);
+
+// Get the Sent items folder
+FolderInfo folderInfo = personalStorage.GetPredefinedFolder(StandardIpmFolder.SentItems);
+
+MessageInfoCollection msgInfoColl = folderInfo.GetContents();
+foreach (MessageInfo msgInfo in msgInfoColl)
+{
+    Console.WriteLine(msgInfo.Subject + ": " + msgInfo.EntryIdString);
+    if (msgInfo.Subject.Equals("some delete condition") == true)
+    {
+        // Delete this item
+        folderInfo.DeleteChildItem(msgInfo.EntryId);
+        Console.WriteLine("Deleted this message");
+    }
+}
+```
+
+### **Deleting Folders from PST Files**
+
+You can delete a PST folder by moving it to the Deleted Items folder.
+
+```csharp
+using (PersonalStorage pst = PersonalStorage.FromFile(@"test.pst"))
+{
+    FolderInfo deletedItemsFolder = pst.GetPredefinedFolder(StandardIpmFolder.DeletedItems);
+    FolderInfo emptyFolder = pst.RootFolder.GetSubFolder("Empty folder");
+	  FolderInfo someFolder = pst.RootFolder.GetSubFolder("Some folder");
+    pst.MoveItem(emptyFolder, deletedItemsFolder);
+	  pst.MoveItem(someFolder, deletedItemsFolder);
+}
+```
+
+The advantage of this method is that the deleted folder can be easily recovered.
+
+```csharp
+FolderInfo someFolder = deletedItemsFolder.GetSubFolder("Some folder");
+pst.MoveItem(someFolder, pst.RootFolder);
+```
+
+You can also permanently remove a folder from the Deleted Items folder, if necessary.
+
+```csharp
+deletedItemsFolder.DeleteChildItem(emptyFolder.EntryId);
+```
+
+The DeleteChildItem method can be used for any folders if you want to immediately and permanently delete subfolder, bypassing the Deleted Items folder.

```

Comment:
`The [DeleteChildItem()](https://apireference.aspose.com/email/net/aspose.email.storage.pst/folderinfo/methods/deletechilditem) method can be used for any folders if you want to immediately and permanently delete subfolder, bypassing the Deleted Items folder.`